hayeon/programmers-2/3차-파일명-정렬.js

```diff
@@ -0,0 +1,24 @@
+function solution(files) {
+  return files.sort((a, b) => {
+    const aHead = a.match(/^\D+/)[0].toLowerCase();
+    const bHead = b.match(/^\D+/)[0].toLowerCase();
+
+    if (aHead < bHead) return -1;
+    if (aHead > bHead) return 1;
+
+    const aNum = a.match(/\d+/)[0].replace(/^0+/, "");
+    const bNum = b.match(/\d+/)[0].replace(/^0+/, "");
+
+    return aNum - bNum;
+  });
+}
```

Comment:
```suggestion
function solution(files) {
  const HEAD_REGEX = /^\D+/;
  const NUMBER_REGEX = /\d+/;

  return files.sort((a, b) => {
    const aHead = a.match(HEAD_REGEX)[0].toLowerCase();
    const bHead = b.match(HEAD_REGEX)[0].toLowerCase();
    const aNum = a.match(NUMBER_REGEX)[0].replace(/^0+/, "");
    const bNum = b.match(NUMBER_REGEX)[0].replace(/^0+/, "");

    return aHead.localeCompare(bHead) || aNum - bNum;
  });
}
```
- sort내부에서 처리하니까 엄청 깔끔하게 풀이가 가능하네요 배워갑니다 :) 
- OR연산자를 이용하면 좀 더 return문을 많이 사용할 필요없이 하나로 끝낼 수 있습니다.
- REGEX같은 경우에는 상수로 빼서 사용하는건 어떨까요?
- 해당 문제는 정규표현식을 사용하지 않고서도 풀 수 있는데 정규표현식을 사용하지 않고 풀이도 진행해보면 좋을 것 같습니다 :)

Reply:
- 감사합니다!! 이문제 좀 어려웠는데 정규표현식 없이도 풀 수 있군요........ 
- 그 방법으로 다시 공부해보겠습니다!!